src/storage/mvcc/reader/reader.rs

```diff
@@ -2442,4 +2448,76 @@ pub mod tests {
         assert_eq!(reader.statistics.write.next, 0);
         assert_eq!(reader.statistics.write.get, 0);
     }
+
+    #[test]
+    fn test_skip_lock_after_upgrade_6_5() {
+        let path = tempfile::Builder::new()
+            .prefix("_test_storage_mvcc_reader_skip_lock_after_upgrade_6_5")
+            .tempdir()
+            .unwrap();
+        let path = path.path().to_str().unwrap();
+        let region = make_region(1, vec![], vec![]);
+        let db = open_db(path, true);
+        let mut engine = RegionEngine::new(&db, &region);
+        let k = b"k";
+
+        // 6.1.0, locks were written
+        let feature_gate = FeatureGate::default();
+        feature_gate.set_version("6.1.0").unwrap();
+        set_tls_feature_gate(feature_gate);
+
+        engine.put(k, 1, 2);
+        // 10 locks were put
+        for start_ts in (6..30).into_iter().step_by(2) {
+            engine.lock(k, start_ts, start_ts + 1);
+        }
+
+        // in 6.5 a new lock was put, and it should contain a `last_change_ts`.
+        let feature_gate = FeatureGate::default();
+        feature_gate.set_version("6.5.0").unwrap();
+        set_tls_feature_gate(feature_gate);
+
+        engine.lock(k, 30, 31);
+        let snap = RegionSnapshot::<RocksSnapshot>::from_raw(db, region);
+        let mut reader = MvccReader::new(snap, None, false);
+        let res = reader
+            .get_write_with_commit_ts(&Key::from_raw(k), 100.into(), None)
+            .unwrap();
+        assert!(res.is_some());
```

Comment:
Better to check if the value is equal, the `DELETE` path should be coverred too.